packages/capacitor-plugin/android/src/main/java/com/capacitorjs/plugins/filetransfer/FileTransferPlugin.kt

```diff
@@ -0,0 +1,190 @@
+package com.capacitorjs.plugins.filetransfer
+
+import android.content.Context
+import com.getcapacitor.JSObject
+import com.getcapacitor.Plugin
+import com.getcapacitor.PluginCall
+import com.getcapacitor.PluginMethod
+import com.getcapacitor.annotation.CapacitorPlugin
+import io.ionic.libs.ionfiletransferlib.IONFLTRController
+import io.ionic.libs.ionfiletransferlib.model.IONFLTRDownloadOptions
+import io.ionic.libs.ionfiletransferlib.model.IONFLTRTransferHttpOptions
+import io.ionic.libs.ionfiletransferlib.model.IONFLTRTransferResult
+import io.ionic.libs.ionfiletransferlib.model.IONFLTRUploadOptions
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
+
+@CapacitorPlugin(name = "FileTransfer")
+class FileTransferPlugin : Plugin() {
+
+    private val ioScope: CoroutineScope by lazy { CoroutineScope(Dispatchers.IO) }
+    private val controller: IONFLTRController by lazy { IONFLTRController() }
+    private lateinit var context: Context
+
+    override fun load() {
+        super.load()
+        context = bridge.context
+    }
+
+    override fun handleOnDestroy() {
+        super.handleOnDestroy()
+        ioScope.cancel()
+    }
+
+    private fun JSObject.toMap(): Map<String, String> {
+        val map = mutableMapOf<String, String>()
+        keys().forEach { key ->
+            map[key] = getString(key) ?: ""
+        }
+        return map
+    }
+
+    private fun JSObject.toParamsMap(): Map<String, Array<String>> {
+        val map = mutableMapOf<String, Array<String>>()
+        keys().forEach { key ->
+            val value = getString(key)
+            if (value != null) {
+                map[key] = arrayOf(value)
+            }
+        }
+        return map
+    }
+
+    @PluginMethod
+    fun downloadFile(call: PluginCall) {
+        val url = call.getString("url") ?: run {
+            call.reject("URL is required")
+            return
+        }
+        val relativePath = call.getString("path") ?: run {
```

Comment:
Minor: Naming-wise, I think it makes more sense to be

```suggestion
        val absolutePath = call.getString("path") ?: run {
```